### DISCUSSION TITLE
ChDb support

### DISCUSSION THREAD

**Discussion by pkese** [DarkWanderer/ClickHouse.Client#555] [General]
It would be nice if ClickHouse.Client supported connecting to a local [ChDb](https://github.com/chdb-io) database.

There doesn't appear to be much code involved for calling into `libchdb`:
https://github.com/chdb-io/chdb-dotnet/blob/6da1be9cb50f01804c2286f16fb290d90e47c3e4/src/chdb/ChDb.cs#L11

ChDb support would be a great addition for people who don't need a full server deployment.

An example use would be for local or CI/CD testing pipelines, so you could run the tests without having to install and configure a full Clickhouse instance.

**Comment by DarkWanderer** [comment]
It would be slightly more involved than that as I essentially need a separate Connection class which - instead of using Https - makes local calls to ChDb. It's not outside the realm of possibility, however, so thanks for the idea - I'll keep it in the backlog
